test/distributed/fsdp/test_fsdp_dtensor_state_dict.py

```diff
@@ -227,6 +228,30 @@ def test_dtensor_sharded_model_load_state_dict(self, offload_to_cpu):
             self.assertEqual(v1, v2)
 
 
+class TestFSDPDeviceMeshInit(DTensorTestBase):
+    @with_comms
+    @skip_if_lt_x_gpu(2)
+    def test_fsdp_device_mesh_init_1d(self):
+        mesh_tensor = torch.arange(self.world_size)
+        device_mesh = DeviceMesh(self.device_type, mesh_tensor)
+        model = FSDP(TestDummyModel().cuda(), _device_mesh=device_mesh)
+        optim = torch.optim.Adam(model.parameters(), lr=0.1)
+        model(model.get_input()).sum().backward()
+
+    @with_comms
+    @skip_if_lt_x_gpu(2)
+    def test_fsdp_device_mesh_init_2d(self):
+        mesh_tensor = torch.arange(self.world_size).view(2, -1)
+        device_mesh = DeviceMesh(self.device_type, mesh_tensor)
+        model = FSDP(
+            TestDummyModel().cuda(),
+            _device_mesh=device_mesh,
```

Comment:
I am hardcoding the mesh dim for HSDP sharded and replicate for now, since we want to quickly roll this part out so modelstore folks can start build the integration for sharded DTensor as well 2D device-mesh support. 

For users, they actually don't have to know the mesh dimension for dp replicate and dp shard, since the default would be dp_replicate: 0, dp_shard: 1. For more advanced users, they can pass an additional arguments in initializing device_mesh. More concrete example is in this diff Chien-Chin has. D47781455